scripts/install/opt/seagate/cortx/motr/libexec/core_logrotate.sh

```diff
@@ -0,0 +1,153 @@
+    PARAM=$1
+    echo "PARAM: $PARAM"
+    if [[ -n "$PARAM" ]]; then
+        retval="OK"
+    else
+        echo "PARAM is empty"
+        retval="continue"
+    fi
+
+    echo $retval
+}
+
+platform=$(get_platform)
+echo "Server type is $platform"
+
+# max core files count in each m0d directory
+core_files_max_count=5
+
+motr_coredirs=("/var/crash/" "/var/log/crash/")
+
+MACHINE_ID=$(cat /etc/machine-id)
+if [ -d "/etc/cortx/motr/sysconfig/$MACHINE_ID" ]; then
+    MOTR_SYSCONFIG="/etc/cortx/motr/sysconfig/$MACHINE_ID/motr"
+else
+    MOTR_SYSCONFIG="/etc/sysconfig/motr"
+fi
+M0TR_M0D_CORE_DIR=$(source "$MOTR_SYSCONFIG" ; echo "$MOTR_M0D_DATA_DIR")
+if [ -n "$M0TR_M0D_CORE_DIR" ]; then
+    motr_coredirs+=("$M0TR_M0D_CORE_DIR")
+fi
+
+while getopts ":n:" option; do
+    case "${option}" in
+        n)
+            core_files_max_count=${OPTARG}
+            if [[ -z "${core_files_max_count}" ]]; then
+              usage
+            fi
+            ;;
+        *)
+            usage
+            ;;
+    esac
+done
+
+if [[ $platform = "virtual" ]]; then
+
+    core_files_max_count=2
+else
+    core_files_max_count=$((core_files_max_count + 2))
+fi
+
+echo "Max core file count: $core_files_max_count"
+echo "Core file directory: $motr_coredirs"
+
+# check for core directory entries
+for motr_coredir in ${motr_coredirs[@]}; do
+    [[ -d $motr_coredir ]] || {
+        echo "$motr_coredir: no such directory"
+        continue
+    }
+
+    [[ $(check_param "$motr_coredir") = "continue" ]] && continue || echo "$motr_coredir"
+
+    # get the core directory of each m0d instance
+    core_dirs=$(find "$motr_coredir" -maxdepth 1 -type d -name m0d-\*)
+
+    [[ $(check_param "$core_dirs") = "continue" ]] && continue || echo "$core_dirs"
+
+    # iterate through all core directories of each m0d instance
+    for core_dir in $core_dirs ; do
+        # get the no. of core file count
+        core_files=$(find "$core_dir" -maxdepth 1 -type f -name "core.*")
+        core_files_count=$(echo "$core_files" | grep -v "^$" | wc -l)
+
+        echo "## found $core_files_count file(s) in core directory $core_dir ##"
+
+        # check core files count is greater than max core file count
+        if [[ "$core_files_count" -gt "$core_files_max_count" ]]; then
+            # get files sort by date - older will come on top
+            remove_file_count=$((core_files_count - core_files_max_count))
+
+            echo "## ($remove_file_count) file(s) can be removed from \
+                           core directory($core_dir) ##"               
+
+            # get the files sorted by time modified 
+            # (most recently modified comes last), that 
+            # is older files comes first
+            
+            if [[ $platform = "physical" ]]; then
+                peserve_files=$((core_files_max_count - 2))
+                files_to_remove=$(ls -tr "$core_dir" | grep core | \
+                                     head -n -"$peserve_files" | awk 'NR>2')
```

Comment:
Why are we considering line numbers greater than 2 only (awk 'NR>2')?

Reply:
have  to preserve only latest 2 core file

Reply:
What is default number of core files that we need to preserve on VM and H/W.  on VM it is 2, and on H/W it is 5 right?

Reply:
incorporated